R/sablefish_om.r

```diff
@@ -295,26 +304,48 @@ for(y in 1:(nyears-1)){
             mort = dp.y$mort[,,1,],
             mat = dp.y$mat[,,1,],
             waa = dp.y$waa[,,1,],
-            sel =  joint_sel,
+            sel =  joint_self,
             ret = joint_ret,
             F = F40,
             avg_rec = mean(recruitment/2)
         )
+        
         ssb <- apply(out_vars$naa_tmp[,,1,]*dp.y$waa[,,1,,drop=FALSE]*dp.y$mat[,,1,,drop=FALSE], 1, sum)
-        hcr_F <- npfmc_tier3a_F(ssb/B40, F35)
-
-        TACs[y+1] <- apply(out_vars$naa_tmp[,,1,]*dp.y$waa[,,1,]*dp.y$mat[,,1,]*joint_sel, 1, sum)*afscOM::F_to_mu(hcr_F)
+        hcr_F[y] <- npfmc_tier3_F(ssb, B40, F40, 0.05)
+        
+        # calculate quantities to get TAC
+        # Project population forward by 1 year with terminal F
+        proj_N <- array(NA, dim = c(dim(out_vars$naa_tmp)))
+        jointsel <- rbind(joint_self, joint_selm)
+    
+        for(s in 1:nsexes) {
```

Comment:
Should do our best to figure out how to do this across both sexes simultaneously rather than looping.